Make the signature of fill_out consistent with fill_.

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#22761 Make the signature of fill_out consistent with fill_.**
* #22760 Move the body of fill_kernel_impl into fill_kernel_cuda
* #22758 Move fill and fill_diagonal to Fill.cpp, Fill.h, and FillKernel.{cpp,cu}

value should be const.

Differential Revision: [D16257779](https://our.internmc.facebook.com/intern/diff/D16257779)

xuhdev committed:
Make the signature of fill_out consistent with fill.

value should be const.

colesbury commented on aten/src/ATen/native/Fill.cpp:
```diff
@@ -16,7 +16,7 @@ Tensor& fill_out(Tensor& self, const Scalar value) {
 }
 
-Tensor& fill_(Tensor& self, Scalar value) {
+Tensor& fill_(Tensor& self, const Scalar value) {
```
We don't usually label value types const

colesbury replied:
(but we're not very consistent about it)

xuhdev replied:
I changed this to consistently not use const

xuhdev committed:
Update on "Make the signature of fill_out consistent with fill."

Make the signature of fill_out consistent with fill.

value should be const.

gh-metadata: pytorch pytorch 22761 gh/xuhdev/12/head

xuhdev committed:
Update on "Make the signature of fill_out consistent with fill."

Make the signature of fill_out consistent with fill.

value should be const.

gh-metadata: pytorch pytorch 22761 gh/xuhdev/12/head

xuhdev commented:
@pytorchbot merge this please
